internal/pkg/config/output.go

```diff
@@ -228,3 +232,50 @@ func makeURL(defaultScheme string, defaultPath string, rawURL string, defaultPor
 	addr.Host = host + ":" + port
 	return addr.String(), nil
 }
+
+func (c *Elasticsearch) DiagRequests(ctx context.Context) []byte {
+	pURL, err := httpcommon.NewProxyURIFromString(c.ProxyURL)
+	if err != nil {
+		zerolog.Ctx(ctx).Warn().Err(err).Msg("Unable to transform proxy_url to url.URL")
+	}
+	settings := httpcommon.HTTPTransportSettings{
+		TLS:     c.TLS,
+		Timeout: c.Timeout,
+		Proxy: httpcommon.HTTPClientProxySettings{
+			Disable: c.ProxyDisable,
+			URL:     pURL,
+			Headers: httpcommon.ProxyHeaders(c.ProxyHeaders),
+		},
+	}
+	headers := http.Header{}
+	for k, v := range c.Headers {
+		headers.Set(k, v)
+	}
+
+	reqs := make([]*http.Request, 0, len(c.Hosts))
+	ctx, cancel := context.WithTimeout(ctx, time.Second*30) // TODO(michel-laterman): Configurable timeout; should this be part of config, or part of a diagnostics action?
```

Comment:
Good question here. Does the `DiagRequest` already send a context with a deadline?

Reply:
Nope, the parent hook is cancelable in case the server needs to shut down

Reply:
The timeout makes the most sense as part of the diagnostics action, that would lead to us exposing it in Fleet when the remote diagnostics request is sent, and not in a totally separate part of the UI users are unlikely to see.

Reply:
I've made https://github.com/elastic/fleet-server/issues/3648 and https://github.com/elastic/elastic-agent-client/issues/113 to track the work needed

Reply:
Until that happens a sensible default should be okay.